src/data_provider/src/ports/portSolarisHelper.cpp

```diff
@@ -0,0 +1,188 @@
+    struct opthdr*       req;
+
+    tor->PRIM_type = T_SVR4_OPTMGMT_REQ;
+    tor->OPT_offset = sizeof (struct T_optmgmt_req);
+    tor->OPT_length = sizeof (struct opthdr);
+    tor->MGMT_flags = T_CURRENT;
+
+    /*
+     * Note: we use the special level value below so that IP will return
+     * us information concerning IRE_MARK_TESTHIDDEN routes.
+     */
+    req = (struct opthdr*)&tor[1];
+    req->level = EXPER_IP_AND_ALL_IRES;
+    req->name  = 0;
+    req->len   = 0;
+
+    ctlbuf.buf = (char*)buf;
+    ctlbuf.len = tor->OPT_length + tor->OPT_offset;
+    flags = 0;
+
+    if (putmsg(sd, &ctlbuf, (struct strbuf*)0, flags) == -1)
+    {
+        return;
+    }
+
+    /*
+     * Each reply consists of a ctl part for one fixed structure
+     * or table, as defined in mib2.h.  The format is a T_OPTMGMT_ACK,
+     * containing an opthdr structure.  level/name identify the entry,
+     * len is the size of the data part of the message.
+     */
+    req = (struct opthdr*)&toa[1];
+    ctlbuf.maxlen = sizeof (buf);
+
+    while (1)
+    {
+        flags = 0;
+        getcode = getmsg(sd, &ctlbuf, (struct strbuf*)0, &flags);
+
+        if (getcode == -1)
+        {
+            return;
+        }
+
+        if (getcode == 0 && ctlbuf.len >= sizeof (struct T_optmgmt_ack) &&
+                toa->PRIM_type == T_OPTMGMT_ACK && toa->MGMT_flags == T_SUCCESS &&
+                req->len == 0)
+        {
+            return;     /* this is EOD msg */
+        }
+
+        if (ctlbuf.len >= sizeof (struct T_error_ack) && tea->PRIM_type == T_ERROR_ACK)
+        {
+            return;
+        }
+
+        if (getcode != MOREDATA || ctlbuf.len < sizeof (struct T_optmgmt_ack) ||
+                toa->PRIM_type != T_OPTMGMT_ACK || toa->MGMT_flags != T_SUCCESS)
+        {
+            return;
+        }
+
+        mib_item_t item
+        {
+            req->level,
+            req->name,
+            req->len,
+            std::shared_ptr<char>(new char[req->len])
+        };
+
+        struct strbuf databuf
+        {
+            item.length,
+            0,
+            item.val.get()
+        };
+
+        flags = 0;
+        getcode = getmsg(sd, (struct strbuf*)0, &databuf, &flags);
+
+        if (getcode != 0)
+        {
+            return;
+        }
+
+        items.push_back(item);
+    }
+}
+
+int PortSolarisHelper::mibOpen(void)
+{
+    int sd;
+
+    sd = open("/dev/arp", O_RDWR);
+
+    if (sd == -1)
+    {
+        perror("arp open");
+        return (-1);
+    }
+
+    if (ioctl(sd, I_PUSH, "tcp") == -1)
+    {
+        perror("tcp I_PUSH");
+        (void) close(sd);
+        return (-1);
+    }
+
+    if (ioctl(sd, I_PUSH, "udp") == -1)
+    {
+        perror("udp I_PUSH");
+        (void) close(sd);
+        return (-1);
+    }
+
+    return (sd);
+}
+
+/* Extract constant sizes */
+void PortSolarisHelper::mibGetConstants(std::vector<mib_item_t>& items)
+{
+    /* 'for' loop 1: */
```

Comment:
What's the meaning for these "for loop 1" comments?

Reply:
original comment will remove!